src/main/java/org/eclipse/yasson/internal/serializer/BigNumberUtil.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2019 Oracle and/or its affiliates. All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 and Eclipse Distribution License v. 1.0
+ * which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ * David Kral
+ ******************************************************************************/
+
+package org.eclipse.yasson.internal.serializer;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * Utility class for checking of IEEE-754 standard in big numbers
+ *
+ * @author David Kral
+ */
+class BigNumberUtil {
+
+    // 53 means max bit value of number with sign bit included
+    private static final int MAX_BIT_SIZE = 53;
+
+    // Max value for Long. Similar to JavaScript Number.MAX_SAFE_INTEGER (2**53)-1
+    private static final long MAX_JS_SAFE_VALUE = 9007199254740991L;
+
+    // Min value for Long. Similar to JavaScript Number.MIN_SAFE_INTEGER -(2**53)+1
+    private static final long MIN_JS_SAFE_VALUE = -9007199254740991L;
+
+    // -1022 is the lowest range of the exponent
+    // more https://en.wikipedia.org/wiki/Exponent_bias
+    private static final int MIN_RANGE = -1022;
+
+    // 1023 is the highest range of the exponent
+    // more https://en.wikipedia.org/wiki/Exponent_bias
+    private static final int MAX_RANGE = 1023;
+
+    /**
+     * Checks whether the value of {@link BigDecimal} matches format IEEE-754
+     *
+     * @param value value which is going to be checked
+     * @return true if value matches format IEEE-754
+     */
+    static boolean isIEEE754(BigDecimal value) {
+        //scale of the number
+        int scale = value.scale();
```

Comment:
The scale here is a "decimal" scale, which cannot be applied to a binary exponent.